These dataset were last accessed on Nov. 13, 2020 at 11:24 am.

In [1]:
import matplotlib.pyplot as plt
import matplotlib
import os
import operator
import pandas as pd

 The dataset was retrieve from __[ML Data](https://www.mldata.io/dataset-details/contraceptive_method/)__, but it can also be found in __[UCI Machine Learning](https://archive.ics.uci.edu/ml/datasets/Contraceptive+Method+Choice)__. This dataset was submitted by Tjen-Sien Lim and was published in 1997. This dataset is a subset of the 1987 National Indonesia Contraceptive Prevalence Survey.

In [7]:
contraceptive_data = pd.read_csv("contraceptive_method_dataset.csv", encoding = "ISO-8859-1", engine='python')

In [8]:
contraceptive_data

wife_age  wife_education  husband_education  children  wife_religion  \
0           24               2                  3         3              1   
1           45               1                  3        10              1   
2           43               2                  3         7              1   
3           42               3                  2         9              1   
4           36               3                  3         8              1   
...        ...             ...                ...       ...            ...   
1468        33               4                  4         2              1   
1469        33               4                  4         3              1   
1470        39               3                  3         8              1   
1471        33               3                  3         4              1   
1472        17               3                  3         1              1   

      wife_working  husband_occupation  standard_of_living_index  \
0                1                   2                         3   
1                1                   3                         4   
2                1                   3                         4   
3                1                   3                         3   
4                1                   3                         2   
...            ...                 ...                       ...   
1468             0                   2                         4   
1469             1                   1                         4   
1470             0                   1                         4   
1471             0                   2                         2   
1472             1                   2                         4   

      media_exposure  contraceptive_method_used  
0                  0                          1  
1                  0                          1  
2                  0                          1  
3                  0                          1  
4                  0                          1  
...              ...                        ...  
1468               0                          3  
1469               0                          3  
1470               0                          3  
1471               0                          3  
1472               0                          3  

[1473 rows x 10 columns]

In [4]:
contraceptive_data.dtypes

wife_age                     int64
wife_education               int64
husband_education            int64
children                     int64
wife_religion                int64
wife_working                 int64
husband_occupation           int64
standard_of_living_index     int64
media_exposure               int64
contraceptive_method_used    int64
dtype: object

_Notes_
- The contraceptive dataset are objects that are in integer form.

In [5]:
contraceptive_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1473 entries, 0 to 1472
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype
---  ------                     --------------  -----
 0   wife_age                   1473 non-null   int64
 1   wife_education             1473 non-null   int64
 2   husband_education          1473 non-null   int64
 3   children                   1473 non-null   int64
 4   wife_religion              1473 non-null   int64
 5   wife_working               1473 non-null   int64
 6   husband_occupation         1473 non-null   int64
 7   standard_of_living_index   1473 non-null   int64
 8   media_exposure             1473 non-null   int64
 9   contraceptive_method_used  1473 non-null   int64
dtypes: int64(10)
memory usage: 115.2 KB


_Notes_
- The contraceptive dataset has 1473 entries and 10 attributes. The attributes are clean which means that all of them have an integer. There are no entries that is na.

In [6]:
attribute_data = pd.read_csv("attributes_contraceptive_method.csv", encoding = "ISO-8859-1", engine='python')
attribute_data

name     type  \
0                   wife_age  integer   
1             wife_education  integer   
2          husband_education  integer   
3                   children  integer   
4              wife_religion  integer   
5               wife_working  integer   
6         husband_occupation  integer   
7   standard_of_living_index  integer   
8             media_exposure  integer   
9  contraceptive_method_used  integer   

                                         description  
0                                                NaN  
1           Categorical. Values: 1=low, 2, 3, 4=high  
2           Categorical. Values: 1=low, 2, 3, 4=high  
3           Number of children ever born (numerical)  
4          Categorical. Values: 0=Non-Islam, 1=Islam  
5                   Categorical. Values: 0=Yes, 1=No  
6                    Categorical. Values: 1, 2, 3, 4  
7           Categorical. Values: 1=low, 2, 3, 4=high  
8                 Binary. Values: 0=Good, 1=Not good  
9  Predictor Class. Values: 1=No-use, 2=Long-term...

_Notes_
- This is a description of the attributes. The wife's age and the number of children are in integer form. The wife's education, husband's education, wife's religion, wife working, husband's occupation, standard of living index, media exposure, and contraceptive method used are categorical values. For clarity, the contraceptive method used has 3 values. The third value is 3 = Short term. The media exposures are binary values with 0 = Good and 1 = Not good. In this study, it is unclear what good and not good means. 